ColorPicker/IkColorPicker.swift

```diff
@@ -0,0 +1,243 @@
+    
+        UIGraphicsEndImageContext();
+    
+        return fullColorImage;
+    }
+    
+    func loadView()
+    {
+        self.frame = CGRect(x:0,y:0,width:320,height:viewPickerHeight);
+        
+        let hueColorImage = self.createHueColorImage();
+        self.hueColorsImageView = UIImageView(image: hueColorImage);
+        self.addSubview(self.hueColorsImageView!);
+        
+        let panRecognizer = UIPanGestureRecognizer(target: self, action:"baseColorPicking:");
+        self.hueColorsImageView!.addGestureRecognizer(panRecognizer);
+        
+        let tapRecognizer = UITapGestureRecognizer(target: self, action: "baseColorPicking:");
+        self.hueColorsImageView!.addGestureRecognizer(tapRecognizer);
+        self.hueColorsImageView!.userInteractionEnabled = true;
+        
+        
+        //------
+        let brightnessColorImage = self.createBrightnessImage(self.selectedBaseColor);
+        self.brightnessColorsImageView = UIImageView(image: brightnessColorImage);
+        self.addSubview(self.brightnessColorsImageView!);
+    
+        var brImgRect = self.brightnessColorsImageView!.frame;
+        brImgRect.origin.y = self.hueColorsImageView!.frame.origin.y + self.hueColorsImageView!.frame.size.height;
+        self.brightnessColorsImageView!.frame = brImgRect;
+        self.brightnessColorsImageView!.userInteractionEnabled = true;
+    
+        let brightSlideGesture = UIPanGestureRecognizer(target: self, action:Selector("colorPicking:"));
+        self.brightnessColorsImageView!.addGestureRecognizer(brightSlideGesture);
+        
+        let brightTapGesture = UITapGestureRecognizer(target: self, action: Selector("colorPicking:"));
+        self.brightnessColorsImageView!.addGestureRecognizer(brightTapGesture);
+        self.brightnessColorsImageView!.userInteractionEnabled = true;
+    }
+    
+    internal var selectedColor: UIColor
+    {
+        didSet{
+            if((self.fullColorImageView) != nil)
+            {
+                let fullColorImage = self.createFullColorImage(selectedColor);
+                fullColorImageView!.image = fullColorImage;
+            }
+            
+            if(self.delegate != nil)
+            {
+                self.delegate!.colorSelectedChanged(self.selectedColor);
+            }
+        }
+    }
+    
+    var selectedBrightness: CGFloat
+    {
+        get
+        {
+            var brightness: CGFloat=0;
+            self.selectedColor.getHue(nil, saturation: nil, brightness:&brightness, alpha:nil);
+
+            return brightness;
+        }
+        
+    }
+    
+    func setSelectedBrightness(brightness: CGFloat)
+    {
+        self.selectedColor = self.colorWithColor(self.selectedBaseColor, brightness:brightness);
+
+    }
+    
+    var selectedBaseColor: UIColor
+    {
+        didSet{
+        var brightnessColorImage = self.createBrightnessImage(self.selectedBaseColor);
+        self.brightnessColorsImageView!.image = brightnessColorImage;
```

Comment:
for type safety, you should probably implicitly unwrap this in the variable definition, or do an if-let right here to make sure it exists. Apple kills a kitten every time you do a force unwrap.